forc-pkg/src/lock.rs

```diff
@@ -151,15 +183,34 @@ impl Lock {
         for pkg in &self.package {
             let key = pkg.name_disambiguated(&disambiguate);
             let node = pkg_to_node[&key[..]];
-            for dep_line in &pkg.dependencies {
+            // If `pkg.contract_dependencies` is None, we will be collecting an empty list of
+            // contract_deps so that we will omit them during edge adding phase
+            let contract_deps: Vec<(String, DepKind)> = pkg
+                .contract_dependencies
+                .clone()
+                .unwrap_or_default()
+                .iter()
+                .map(|contract_dep| (contract_dep.clone(), DepKind::Contract))
+                .collect();
+            // If `pkg.dependencies` is None, we will be collecting an empty list of
+            // lib_deps so that we will omit them during edge adding phase
+            let lib_deps: Vec<(String, DepKind)> = pkg
+                .dependencies
+                .clone()
+                .unwrap_or_default()
+                .iter()
+                .map(|lib_dep| (lib_dep.clone(), DepKind::Library))
+                .collect();
+            for (dep_line, dep_kind) in lib_deps.iter().chain(contract_deps.iter()) {
```

Comment:
Another way of writing this section might be:

```rust
            let contract_deps: Vec<(String, DepKind)> = pkg
                .contract_dependencies
                .as_ref()
                .into_iter()
                .flat_map(|vec| vec)
                .map(|contract_dep| (contract_dep, DepKind::Contract));
            let lib_deps: Vec<(String, DepKind)> = pkg
                .dependencies
                .as_ref()
                .into_iter()
                .flat_map(|vec| vec)
                .map(|lib_dep| (lib_dep, DepKind::Library));
    for (dep_line, dep_kind) in lib_deps.chain(contract_deps) {
```
^ This should achieve the same behaviour avoiding both the `Vec` clones and the `String` clones.

Reply:
Thanks! I can address this in the next PR to not to dismiss your approval 👍